controller/lib/core/alarm.cpp

```diff
@@ -12,11 +12,18 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-#include <Arduino.h>
+#include "hal.h"
 
 #include "alarm.h"
 
-static stack_t stack;
+namespace {
+struct alarm_stack_t {
```

Comment:
Outside this PR of course, but this file looks more like C than C++ - is it because we're not allowed to allocate memory on the controller or something like that?

Reply:
Correct, we don't have malloc.

Nonetheless I think it would be an improvement to C++'ify a lot of our code.

Reply:
(I used the anonymous namespace even though this is towards the "C" side of the spectrum because there's actually no way in C to make a type that's private to a file.)